### PR TITLE
doc(samples): Add a readme for device samples

### PR DESCRIPTION
**Pull request by abhipsaMisra**
Link to readme: https://github.com/Azure/azure-iot-sdk-csharp/blob/abmisr/sampleReadme/iothub/device/samples/readme.md

### CODE REVIEW

File: iothub/device/samples/readme.md [hunk 1]
```diff
@@ -1,27 +1,71 @@
...
 ### [Module samples][module-samples]
-* [MessageSample][m-message-sample]
-* [TwinSample][m-twin-sample]
+* [Message Sample][m-message-sample]
+* [Twin Sample][m-twin-sample]
+
+### Prerequisites
+In order to run the device samples on Linux or Windows, you will first need the following prerequisites:
+* [Setup your IoT hub][lnk-setup-iot-hub]
+* [Provision your device and get its credentials][lnk-manage-iot-hub]
```
Comment (bikamani) [inline iothub/device/samples/readme.md:21]: why manage hub link?
It should be this one right? https://github.com/Azure/azure-iot-device-ecosystem/blob/master/setup_iothub.md#retrieving-user-credentials-to-interact-with-the-service-not-as-a-device

File: iothub/device/samples/readme.md [hunk 2]
```diff
@@ -1,27 +1,71 @@
...
+
+### Setup environment
+
+The following prerequisite is the minimum requirement to build and run the samples. 
+
+Visual Studio is **not required** to run the samples.
+
+- Install the latest .NET Core from https://dot.net
```
Comment (bikamani) [inline iothub/device/samples/readme.md:29]: I would say, point to samples doc repo here

Change that doc to call latest .net core

File: iothub/device/samples/readme.md (hunk 1 above)
Comment (abhipsaMisra) [reply iothub/device/samples/readme.md:21]: this is device sample, so it needs to interact as a device, not service

File: iothub/device/samples/PnpDeviceSamples/readme.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+### Plug and Play Samples
+
+These samples demonstrate how a plug and play enabled device interacts with IoT hub, to:
+- Send telemetry.
+- Update read-only and read-write porperties.
+- Respond to command invocation. 
+
+The samples demonstrate two scenarios:
+- The device is modeled as a plug and play device, having only a root interface - [Thermostat][d-thermostat]
+  - This model defines root level telemetry, read-only and read-write properties and commands.:
```
Comment (bikamani) [inline iothub/device/samples/PnpDeviceSamples/readme.md:10]: remove ":" from the end

File: iothub/device/samples/readme.md (hunk 2 above)
Comment (abhipsaMisra) [reply iothub/device/samples/readme.md:29]: the samples repo is not maintained by us, so I don't want to add a link to those instructions. Its a single line instruction anyway

File: iothub/device/samples/PnpDeviceSamples/readme.md (hunk 3 above)
Comment (abhipsaMisra) [reply iothub/device/samples/PnpDeviceSamples/readme.md]: good catch, will do

File: iothub/device/samples/readme.md (hunk 2 above)
Comment (bikamani) [reply iothub/device/samples/readme.md:29]: Don't we point our customers to samples repo currently?

File: iothub/device/samples/readme.md (hunk 1 above)
Comment (bikamani) [reply iothub/device/samples/readme.md:21]: The link you currently have does not seem to point how to provision device or get credentials

File: iothub/device/samples/PnpDeviceSamples/readme.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+### Plug and Play Samples
+
+These samples demonstrate how a plug and play enabled device interacts with IoT hub, to:
+- Send telemetry.
+- Update read-only and read-write porperties.
+- Respond to command invocation. 
+
+The samples demonstrate two scenarios:
+- The device is modeled as a plug and play device, having only a root interface - [Thermostat][d-thermostat]
```
Comment (bikamani) [inline iothub/device/samples/PnpDeviceSamples/readme.md:9]: I would call out the sample name in beginning with a link to interface followed at the end both here and on line 11

File: iothub/device/samples/PnpDeviceSamples/readme.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  - This model defines root level telemetry, read-only and read-write properties and commands.
+- The device is modeled as a plug and play device having multiple components - [Temperature controller][d-temperature-controller].
+  - This model defines root level telemetry, read-only property and commands.
+  - It also defines two [Thermostat][d-thermostat] components, and a [device information][d-device-info] component.
```
Comment (bikamani) [inline iothub/device/samples/PnpDeviceSamples/readme.md:13]: no need to add a link to thermostat sample. If we really want to show component then I would add a link to dtdl showing this component as part of the model

File: iothub/device/samples/readme.md (hunk 2 above)
Comment (abhipsaMisra) [reply iothub/device/samples/readme.md:29]: for pre-requisites/ setup instructions? no, we don't.

File: iothub/device/samples/PnpDeviceSamples/readme.md (hunk 5 above)
Comment (abhipsaMisra) [reply iothub/device/samples/PnpDeviceSamples/readme.md:13]: the thermostat sample has the model json file

Comment (bikamani) [reply iothub/device/samples/PnpDeviceSamples/readme.md:13]: sure but that doesn't help in this comment

Comment (abhipsaMisra) [reply iothub/device/samples/PnpDeviceSamples/readme.md:13]: I can add a link to the json file in the thermostat sample, sure

File: iothub/device/samples/PnpDeviceSamples/readme.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+- The device is modeled as a plug and play device, having only a root interface - [Thermostat][d-thermostat]
+  - This model defines root level telemetry, read-only and read-write properties and commands.
+- The device is modeled as a plug and play device having multiple components - [Temperature controller][d-temperature-controller].
+  - This model defines root level telemetry, read-only property and commands.
+  - It also defines two [Thermostat][thermostat-model] components, and a [device information][d-device-info] component.
+
+[d-thermostat]: https://github.com/Azure/azure-iot-sdk-csharp/tree/master/iothub/device/samples/PnpDeviceSamples/Thermostat
+[d-temperature-controller]: https://github.com/Azure/azure-iot-sdk-csharp/tree/master/iothub/device/samples/PnpDeviceSamples/TemperatureController
+[thermostat-model]: /iothub/device/samples/PnpDeviceSamples/Thermostat/Models/Thermostat.json
+[d-device-info]: https://devicemodels.azureiotsolutions.com/models/public/dtmi:azure:DeviceManagement:DeviceInformation;1?codeView=true
```
Comment (bikamani) [inline iothub/device/samples/PnpDeviceSamples/readme.md:18]: Just wondering - if this is where we should be pointing?

File: iothub/device/samples/readme.md (hunk 2 above)
Comment (bikamani) [reply iothub/device/samples/readme.md:29]: not for pre requisite but that's been where they go for samples anyways. We should have one place for pre-req as those are not different device/service samples and point to it in our docs here

Comment (abhipsaMisra) [reply iothub/device/samples/readme.md:29]: sure, you can add that in you PR, go for it!

**Pull request merged**